docs/source/how_to_guides/adding_a_new_model.rst

```diff
@@ -0,0 +1,198 @@
+
+.. code-block:: bash
+
+    ./anomalib/models/<new-model>
+    ├── __init__.py
+    ├── config.yaml
+    ├── torch_model.py
+    ├── lightning_model.py
+    ├── loss.py    # OPTIONAL
+    ├── anomaly_map.py    # OPTIONAL
+    └── README.md
+
+
+Create a ``config.yaml`` file
+-------------------------------
+
+This file stores all the configurations, from data to optimization options. An exemplary yaml file is shown below. For a full configuration file, you could refer to one of the existing model implementations, such as `patchcore configuration. <https://github.com/openvinotoolkit/anomalib/blob/main/anomalib/models/patchcore/config.yaml>`_
+
+.. code-block:: yaml
+
+    dataset:
+        name: mvtec #options: [mvtec, btech, folder]
+        format: mvtec
+        ...
+    model:
+        name: patchcore
+        backbone: wide_resnet50_2
+        ...
+    metrics:
+        image:
+            - F1Score
+        ...
+    visualization:
+        show_images: False # show images on the screen
+        ...
+    # PL Trainer Args. Don't add extra parameter here.
+    trainer:
+        accelerator: auto # <"cpu", "gpu", "tpu", "ipu", "hpu", "auto">
+        ...
+
+
+
+Create a ``torch_model.py`` Module
+----------------------------------
+This file contains the torch model implementation that inherits from ``torch.nn.Module`` and performs a basic forward-pass. The advantage of storing the model in a separate ``torch_model.py`` file is that the model is de-coupled from the rest of the anomalib implementations and could be used outside the library as well. Basic implementation would look like as follows:
+
+.. code-block:: python
+
+    class NewModelModel(nn.Module):
+        """New Model Module."""
+        def __init__(self):
+            pass
+        def forward(self, x):
+            pass
+
+
+Create a ``lightning_model.py`` Module
+--------------------------------------
+This module contains the actual lightning model implementation that inherits from the ``AnomalModule``, which already has the ``anomalib`` related attributes and methods. The user does not need to worry about the boilerplate code and only needs to implement the training and validation logic of the algorithm.
+
+.. code-block:: python
+
+    class NewModel(AnomalyModule):
+        """PL Lightning Module for the New Model."""
+        def __init__(self):
+            super().__init__()
+            pass
+        def training_step(self, batch):
+            pass
+        ...
+        def validation_step(self, batch):
+            pass
+
+Create a ``loss.py`` File - [Optional]
+--------------------------------------
+This module's availability is dependent on the algorithm. If the algorithm requires a custom loss function, this file contains the subclass of the torch.nn.Module class implementation. This loss would subsequently be utilized by the lightning module.
```

Comment:
~~this file contains the~~ this file may contain the

If the custom loss function is very simple, it might not be needed to have a separate class for this.